code/modules/shuttle/helpers.dm

```diff
@@ -0,0 +1,134 @@
+		return single_controller.is_locked
+	else
+		WARNING("Direction [direction] does not exist.")
+	return FALSE
+
+/datum/door_controller/aggregate/proc/control_doors(action, direction, force)
+	if(direction == "all")
+		if(allow_multicast)
+			for(var/door_group in door_controllers)
+				var/datum/door_controller/single/controller = door_controllers[door_group]
+				controller.control_doors(action, force)
+			return
+		else
+			WARNING("Aggregate door controller does not support multicast.")
+	if(door_controllers[direction])
+		var/datum/door_controller/single/controller = door_controllers[direction]
+		controller.control_doors(action, force)
+	else
+		WARNING("Door group [direction] does not exist.")
+
+/datum/door_controller/aggregate/proc/get_data()
+	. = list()
+	var/all_locked = TRUE
+
+	for(var/direction in door_controllers)
+		var/datum/door_controller/single/controller = door_controllers[direction]
+		var/list/door_data = list("id" = direction, "value" = controller.is_locked)
+		. += list(door_data)
+		if(!controller.is_locked)
+			all_locked = FALSE
+
+	var/list/door_data = list("id" = "all", "value" = all_locked)
+	. += list(door_data)
+
+/**
+	Datum containing methods to allow for the control over a single door group.
+*/
+/datum/door_controller/single
+	var/label = "dropship"
+	var/list/doors = list()
+	var/is_locked = FALSE
+
+/datum/door_controller/single/Destroy(force, ...)
+	. = ..()
+	doors = null
+
+/datum/door_controller/single/proc/control_doors(action, force = FALSE)
+	for(var/D in doors)
+		var/obj/structure/machinery/door/door = D
+		var/is_external = door.borders_space()
+		// do not allow the user to normally control external doors
+		if(!force && is_external)
+			continue
+		switch(action)
+			if("open")
+				INVOKE_ASYNC(door, TYPE_PROC_REF(/obj/structure/machinery/door/airlock, open))
+			if("close")
+				INVOKE_ASYNC(door, TYPE_PROC_REF(/obj/structure/machinery/door/airlock, close))
+			if("force-lock")
+				INVOKE_ASYNC(src, PROC_REF(lockdown_door), door)
+				is_locked = TRUE
+			if("lock")
+				INVOKE_ASYNC(door, TYPE_PROC_REF(/obj/structure/machinery/door/airlock, lock))
+				is_locked = TRUE
+			if("unlock")
+				INVOKE_ASYNC(door, TYPE_PROC_REF(/obj/structure/machinery/door/airlock, unlock))
+				is_locked = FALSE
+			if("force-lock-launch")
+				INVOKE_ASYNC(src, PROC_REF(lockdown_door_launch), door)
+				is_locked = TRUE
+			if("force-unlock")
+				INVOKE_ASYNC(src, PROC_REF(force_lock_open_door), door)
+				is_locked = FALSE
+
+/datum/door_controller/single/proc/lockdown_door_launch(obj/structure/machinery/door/airlock/air)
+	var/list/door_turfs = get_turf(air)
+	if(istype(air, /obj/structure/machinery/door/airlock/multi_tile))
+		var/obj/structure/machinery/door/airlock/multi_tile/multi_door = air
+		door_turfs = multi_door.get_filler_turfs()
+	for(var/turf/door_turf in door_turfs)
+		for(var/mob/blocking_mob in door_turf.loc) // Bump all mobs outta the way for outside airlocks of shuttles
```

Comment:
```suggestion
		for(var/mob/blocking_mob in door_turf) // Bump all mobs outta the way for outside airlocks of shuttles
```
`Door_turf` is a turf. Its `.loc` is the entire area, so everybody gets kicked out.